### PR TITLE
Fixed possible buffer overflow on system privileges lookup during user mapping

### PR DESCRIPTION
**Pull request by dmitry-starodubov**

### CODE REVIEW

File: src/jrd/Mapping.cpp [hunk 1]
```diff
@@ -1171,7 +1171,9 @@ class SysPrivCache : public PermanentStorage
 					g |= l;
 				}
 
-				ULONG gg = 0; g.store(&gg);
+				FB_UINT64 gg = 0;
+				fb_assert(sizeof(gg) >= g.BYTES_COUNT);
```
Comment (aafemt) [inline src/jrd/Mapping.cpp:1175]: `static_assert` would be better here because none side of the comparison can be changed in runtime.

Comment (dmitry-starodubov) [reply src/jrd/Mapping.cpp]: done

**Comment by AlexPeshkoff** [comment]
Why FB4 only? It makes no sense to change old branch and leave fresh one as is. Such change will be lost.

Please provide it at least for master, and better for FB5 too.

**Comment by dmitry-starodubov** [comment]
They were also made yesterday: #8125 and #8126

**Comment by aafemt** [comment]
In my comment I didn't pay attention to target branch assuming that PRs are made for master at the first hand. Is static_assert allowed for usage in v4 branch?

**Comment by dyemanov** [comment]
Yes, it's allowed.

**Comment by AlexPeshkoff** [comment]
On 5/24/24 14:35, Dimitry Sibiryakov wrote:

> Is static_assert allowed for usage in v4 branch?
>

Yes.

**Pull request merged**